### PR TITLE
Set up basic structure

### PR DESCRIPTION
**Pull request by Bibo-Joshi**
Included  `extract_url` helpers as first contribution as template/show case

some remarks/discussion points/open questions:

* ~I feel like naming the package `ptb_contrib` but the repo `contrib` is inconsistent. Can we still change that?~ ✔️ 
* For now I put as copyright
    ```python
    # A library containing community-based extension for the python-telegram-bot library
    # Copyright (C) 2020
    # The ptb-contrib developers
    ```
* test setup is at a bare minimum. The bot that's provided has a fake token and can't be acutally used. I don't think that contributions will actually need to run valid tests with TG but can just monkeypatch everything. If they do need it at some point, we can re-add …
* ~also tests apparently only run once the workflow files are merged into master?!~ Nah, I just didn't remember that we now call it `main` instead of `master` …
* `setup.py` automatically inferes any extra requirements for each module by reading `ptbcontrib/module/requirements.txt`. I don't know if such a setup goes against any conventions. But each module having it's own requirements file (if needed) seemed like a clean approach to me
* `setup.py` current has
    ```python
          author_email='devs@python-telegram-bot.org',
          url='https://python-telegram-bot.org/',
    ```
    Or should that be changed?
* is it worth it linking this repo on the homepage? e.g. `Extensions` next to `Download`?
* ~@JosXa proposes a monorepo approach in #2~
* We might get in trouble with the tests at some point, if different modules have clashing requirements. like `somelib<=7.0` and `somelib>=8.0`. However IMHO
    * we don't need to worry about that as long as it doesn't happen
    * the module requirering an old version of the lib should just be updated or deprecated in that case.
* About Codacy: We'll want to ignore the `tests` and `.github` directories, but I can only ignore them once they're on `main`

**TODO:**
- [x] Add a README to `extracturls`
- [x] Add note about adding a README to contributions in the contribution guide

### CODE REVIEW

**Comment by JosXa** [comment]
Seems it's time to leave a #FIRST! 😜 With both first comment and first feature, haha



Good stuff. I'll have a look and review this week.

**Review by Poolitzer — CHANGES_REQUESTED** [review]
some slight changes

File: .github/CONTRIBUTING.rst [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+       self.last_name = last_name
+
+    # BAD
+    def __init__(self, id, name, last_name=None):
+       self.last_name = last_name
+
+
+.. _`Code of Conduct`: https://www.python.org/psf/codeofconduct/
+.. _`issue tracker`: https://github.com/python-telegram-bot/ptbcontrib/issues
+.. _`Telegram group`: https://telegram.me/pythontelegrambotgroup
```
[the remark this answers is not in the document]
Comment (Bibo-Joshi) [reply .github/CONTRIBUTING.rst:223]: I'm not sure if opening another group just for ptbcontrib is worth it and it's not really off-topic. I guess having ptbcontrib questions in the on-topic group is not too bad. We can always change that, though …

File: .github/CONTRIBUTING.rst [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+It's always good to not initialize optional arguments at class creation,
+instead use ``**kwargs`` to get them. It's well known Telegram API can
+change without notice, in that case if a new argument is added it won't
+break the API classes. For example:
+
+.. code-block:: python
+
+    # GOOD
+    def __init__(self, id, name, last_name=None, **kwargs):
```
[the remark this answers is not in the document]
Comment (Bibo-Joshi) [reply .github/CONTRIBUTING.rst:213]: True. I just removed the paragraph …

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+- `Contributing`_
+
+- `License`_
+
+============
+Introduction
+============
+
+This library provides extensions for the `python-telegram-bot <https://python-telegram-bot.org>`_ library written and maintained by the community of PTB users.
```
[the remark this answers is not in the document]
Comment (Bibo-Joshi) [reply README.rst:58]: Do we have multiple cumminities? o.0

File: README.rst [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+.. code:: shell
+
+    $ pip install git+https://github.com/python-telegram-bot/ptbcontrib.git
+
+If you want to use an extension that has some special requirements, you can install them on the fly as e.g.
+
+.. code:: shell
+
+    $ pip install "ptbcontrib[extension1,extension2] @ git+https://github.com/python-telegram-bot/ptbcontrib.git"
```
[the remark this answers is not in the document]
Comment (Bibo-Joshi) [reply README.rst:74]: you think so? We could put `extract_urls` there, but I don't see a huge benefit. anyone who's not able to see that those are placeholders is probably not ready to use ptbcontrib …

File: requirements-dev.txt [hunk 5]
```diff
@@ -0,0 +1,11 @@
+flake8
+pep257
+pylint
```
[the remark this answers is not in the document]
Comment (Bibo-Joshi) [reply requirements-dev.txt:3]: Well, `pre-commit` installs them on it's own, but if you want to run them manually, you need to have them installed. But, a few dev-reqs don't hurt too much, I guess. Switching yapf for black, though …

**Comment by Bibo-Joshi** [comment]
Thanks for the comments @JosXa @Poolitzer :) Updated a bit and left some replies

File: .github/CONTRIBUTING.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (Poolitzer) [reply .github/CONTRIBUTING.rst:192]: okay, lets see how it grows, we might want to outsource it then

File: README.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (Poolitzer) [reply README.rst:58]: potentially. This seems so narrowed down like you already have a defined one, and well, we dont :D

File: README.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (Poolitzer) [reply README.rst:74]: hm, okay, fair enough.

File: requirements-dev.txt (hunk 5 above)
[the remark this answers is not in the document]
Comment (Poolitzer) [reply requirements-dev.txt:3]: ha at least that

**Review by Poolitzer — COMMENTED** [review]
LGTM

File: README.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (Bibo-Joshi) [reply README.rst:58]: From my understanding "the community of PTB users" just means th set of all humans that use PTB. I don't see a benefit of differentiating between different subsets of that

**Pull request merged**